File: M2 S3/PrePost/1.py
import gmsh
import sys

gmsh.initialize()
gmsh.model.add("square")

Lc = 1e-1
Lx = Ly = 1.0

p1 = gmsh.model.geo.addPoint(-Lx, -Ly, 0, Lc)
p2 = gmsh.model.geo.addPoint(Lx, -Ly, 0, Lc)
p3 = gmsh.model.geo.addPoint(Lx, Ly, 0, Lc)
p4 = gmsh.model.geo.addPoint(-Lx, Ly, 0, Lc)

l1 = gmsh.model.geo.addLine(p1, p2)
l2 = gmsh.model.geo.addLine(p2, p3)
l3 = gmsh.model.geo.addLine(p3, p4)
l4 = gmsh.model.geo.addLine(p4, p1)

loop = gmsh.model.geo.addCurveLoop([l1, l2, l3, l4])
surface = gmsh.model.geo.addPlaneSurface([loop])

gmsh.model.geo.synchronize()
gmsh.model.mesh.generate(2)

if '-nopopup' not in sys.argv:
    gmsh.fltk.run()

gmsh.finalize()

File: M2 S3/PrePost/1.cpp
#include <set>
#include <gmsh.h>
int main(int argc, char **argv)
{
  gmsh::initialize();

  gmsh::model::add("square");
  gmsh::model::geo::addPoint(.1, 0, 0, lc, 2);
  
  roundedRadius = 0;
  tag = -1;
  gmsh::model::geo::addRectangle(-0.5, -0.5, -0.5, 1, 1, tag, roundedRadius);

  gmsh::model::geo::synchronize();
  gmsh::model::addPhysicalGroup(2, {1}, -1, "My surface");

  std::set<std::string> args(argv, argv + argc);
  if(!args.count("-nopopup")) gmsh::fltk::run();
  
  gmsh::finalize();
  return 0;
}